// Copyright 2023 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package privacy_sandbox.server_common;

import "apis/privacysandbox/apis/roma/app_api/v1/options.proto";

// For faster allocations of sub-messages.
option cc_enable_arenas = true;

service TestService {
  option (privacysandbox.apis.roma.app_api.v1.roma_svc_annotation) = {
    name: 'Test Server',
    code_id: "app_api_roma_app_test_v1",
    description: 'Test Server',
    cpp_namespace: 'privacy_sandbox::server_common',
    roma_app_name: 'TestServer',
  };

  rpc TestMethod(TestMethodRequest) returns (TestMethodResponse) {
    option (privacysandbox.apis.roma.app_api.v1.roma_rpc_annotation) = {
      description: 'TestMethodAnnotation'
    };
  };
}

message TestMethodRequest {
  string input = 1;
  int32 processing_delay_ms = 2;
}

message TestMethodResponse {
  string output = 1;
}
